modules/core/SolidSphere.cpp

```diff
@@ -0,0 +1,42 @@
+#include "SolidSphere.h"
+#include "PrimTriangle.h"
+
+
+rt::CSolidSphere::CSolidSphere(rt::ptr_shader_t pShader, const Vec3f &origin, float radius, size_t sides) : CSolid(origin)
+{
+    m_origin = origin;
+    m_radius = radius;
+    m_sides = sides;
+
+    // This is quite straight forwards - The function creates 4 points and creates a quad from them
+    // It uses UV coordinates to so. Except at the cap where it creates triangles from the "north/south poles"
+    // to the points from a layer below.
+    for (int i = 0; i <= sides; i++)
```

Comment:
the number of vertical segments should be equal to half of the sides segments. Thus, if there are n sides segments we have:
n triangles on top
n triangles at the bottom
(n / 2 -2) * n quads of the body, i.e. (n / 2 - 2) * n * 2 triangles.
Thus for n sides we get
2 * n + n * n - 4 * n = (n^2 - 2*n)  triangles. (for test)

Reply:
The minimum number of sides is 4. It is possible to add test, where someone tries to init sphere with 3 sides

Reply:
Does it make sense to add a height parameter in the function where the user can input the amount of lines they want height wise?

Reply:
👍  RT_ASSERTION to check if the number of sides is more than 3 otherwise display message to user. Sample error:
![image](https://user-images.githubusercontent.com/41652846/100075844-c2727e00-2e40-11eb-979d-156ff4df3a22.png)

👍  Adjusted logic to follow the formula for the total amount of PrimTri's.